crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -80,13 +74,6 @@ impl Settlement {
                 score,
             );
         }
-        if score < promised_solution.score() {
-            return Err(Error::ScoreMismatch {
-                expected: promised_solution.score(),
-                got: score,
-            })
-            .map_err(with(auction_id));
-        }
```

Comment:
Removed as suggested here: https://github.com/cowprotocol/services/pull/2861#discussion_r1705224373

Still dependent on resolving the https://github.com/cowprotocol/services/issues/2797 as apparently we still have some rounding issues.